### PR TITLE
Add `luarocks` package type

### PR DESCRIPTION
**Pull request by hishamhm**
Hi, I am the lead mantainer of the [LuaRocks](https://luarocks.org) package manager and we'd like to add its entry to the `purl` spec. Thanks!

### CODE REVIEW

File: PURL-TYPES.rst [hunk 1]
```diff
@@ -311,6 +311,22 @@ huggingface
       pkg:huggingface/microsoft/deberta-v3-base@559062ad13d311b87b2c455e67dcd5f1c8f65111?repository_url=https://hub-ci.huggingface.co
 
 
+luarocks
+--------
+``luarocks`` for Lua packages installed with LuaRocks:
+
+- ``namespace``: The user manifest under which the package is registered. If not given, the root manifest is assumed.
```
Comment (stevespringett) [inline PURL-TYPES.rst:318]: Is the namespace case sensitive or insensitive? Please specify which.

File: PURL-TYPES.rst [hunk 2]
```diff
@@ -311,6 +311,22 @@ huggingface
...
+- ``name``: The LuaRocks package name.
```
Comment (stevespringett) [inline PURL-TYPES.rst:319]: Is the name case sensitive or insensitive? Please specify which.

File: PURL-TYPES.rst [hunk 3]
```diff
@@ -311,6 +311,22 @@ huggingface
...
+- ``version``: The full LuaRocks package version, including module version and rockspec revision.
```
Comment (stevespringett) [inline PURL-TYPES.rst:320]: Is the version case sensitive or insensitive? Please specify which. Additionally, is the version required?

**Review by stevespringett — CHANGES_REQUESTED** [review]

**Comment by hishamhm** [comment]
@stevespringett Added all the requested clarifications!

**Pull request merged**